src/resources/views/crud/fields/relationship.blade.php

```diff
@@ -0,0 +1,53 @@
+<!--  relationship  -->
+
+@php
+    use Illuminate\Support\Str;
+
+    if(isset($field['inline_create']) && !is_array($field['inline_create'])) {
+        $field['inline_create'] = [true];
+    }
+    //dd($crud);
+
+    $field['multiple'] = $field['multiple'] ?? false;
+    $field['ajax'] = $field['ajax'] ?? false;
+    $field['placeholder'] = $field['placeholder'] ?? ($field['multiple'] ? 'Select entries' : 'Select entry');
+    $field['allows_null'] = $field['allows_null'] ?? $crud->model::isColumnNullable($field['name']);
+    // Note: isColumnNullable returns true if column is nullable in database, also true if column does not exist.
+
+    // if field is not ajax but user wants to use InlineCreate
+    // we make minimum_input_length = 0 so when user open we show the entries like a regular select
+    $field['minimum_input_length'] = ($field['ajax'] !== true) ? 0 : ($field['minimum_input_length'] ?? 2);
+
+   @endphp
+
+   @if(isset($field['inline_create']))
+        @include('crud::fields.relationship.fetch_or_create')
+    @else
+        @if($field['ajax'])
+            @include('crud::fields.fetch')
+        @else
+            @include('crud::fields.relationship.select')
+        @endif
+   @endif
+
+
+    @if ($crud->fieldTypeNotLoaded($field))
+        @php
+            $crud->markFieldTypeAsLoaded($field);
+        @endphp
+
+        {{-- FIELD CSS - will be loaded in the after_styles section --}}
+        @push('crud_fields_styles')
+
+
+        @endpush
+
+        {{-- FIELD JS - will be loaded in the after_scripts section --}}
+        @push('crud_fields_scripts')
+        @stack('crud_fields_scripts')
```

Comment:
## Relationship Issue 9

I see this ```@stack('crud_fields_scripts')``` is being pushed inside the ```crud_fields_scripts``` stack. So... inside itself?! I know you've already explained this to me but could you please explain again how this works, how the additional assets (for the modal) are loaded onpage?

The reason I noticed this is because this stack is also defined (once more) inside ```src/resources/views/crud/fields/relationship/show_fields.blade.php``` - are you sure it's needed in both places?

What I don't quite understand is how this Blade stack can be defined 3 times (once by the regular Backpack, once by relationship, and once by relationship/show_fields) and everything STILL WORKS 😆 And why relationship/show_tabbed_fields doesn't have them - if one did I expected the other to need them too.

Sorry for being thick about this 😬 🙄

Reply:
Hahah indeed, this is a bit confusing since the beginning. Even for me is(was) confusing , I can imagine your pain. I am sorry :see_no_evil: 

The stacks are only required in `relationship/show_fields.blade.php`

I just cleaned everything up and added an inline coment explaining how they work. Is as follows:

```html
<!--  This is where modal fields assets are pushed.
        We bind the modal content including what fields pushed to this stacks to the end of the body tag,
        so we make sure all backpack assets are previously loaded, like jquery etc.

        We can give the same stack name because backpack `crud_fields_scripts` is already rendered and
        this is the only available when rendering the modal html.
    -->
```

Reply:
Ooooh I get it now. So basically we don't mind calling it ```crud_fields_scripts``` because it's already loaded, so it doesn't matter. Now I understand. Thank you for your patience 😁